manifests/utility/database.pp

```diff
@@ -0,0 +1,24 @@
+# Sample Usage:
+#   mariadb::utility::database { "production dbname" :
+#       dbname => 'dbname',
+#       charset => 'utf8',
+#       collate => 'utf8_general_ci'
+#       ensure => 'present',
+#   }
+define mariadb::utility::database($dbname, $charset = 'utf8', $collate = 'utf8_general_ci', $ensure = 'present' ) {
```

Comment:
What is the advantage of using these utility defined types vs just using the resources from the mysql module directly? What problem is this solving for you?